### PR TITLE
Feature/improve database

### PR DESCRIPTION
**Pull request by otahontas**
### Description
Another small update, for database this time. 

### Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Changes Made
- Add missing crud operations (replace, update, delete)
- Add tests for crud operations
- Fix bug, where mongoclient was created individually each db_service object. MongoClient should be shared accross the whole application, since client handles pooling: [see more here](https://pymongo.readthedocs.io/en/stable/faq.html#how-does-connection-pooling-work-in-pymongo)

### Testing
- [x] Unit Tests

### CODE REVIEW

File: metadata_backend/database/db_services.py [hunk 1]
```diff
@@ -1,37 +1,35 @@
...
+found from deploy/mongodb.
+"""
 
 import os
 from typing import Dict
 
 from pymongo import MongoClient, errors
 from pymongo.cursor import Cursor
 
+# Set up database client
```
Comment (blankdots) [inline metadata_backend/database/db_services.py:17]: how about we have a config module, to properly manage config ?

File: metadata_backend/database/db_services.py [hunk 2]
```diff
@@ -77,3 +74,58 @@ def read(db_service: DBService, collection: str,
...
+        :param accession_id: Accession id for object to be updated
+        :param data_to_be_updated: JSON representing the data that replaces
+        old data
+        :raises: Error when read fails for any Mongodb related reason
+        """
+        try:
+            find_by_id_query = {"accessionId": accession_id}
+            db_service.database[collection].replace_one(find_by_id_query,
+                                                        data_to_be_updated)
+        except errors.ConnectionFailure:
```
Comment (blankdots) [inline metadata_backend/database/db_services.py:114]: I think we should catch more that `ConnectionFailure`, maybe catch the specific ones first and then catch the general exception.

should be the same in `delete`.

also on `ConnectionFailure`, one should do a retry mechanism to retry the previous operation e.g. 10 times in 2-5 min after which raise an error (probably could be a later task -- add an issue if so).

File: metadata_backend/database/db_services.py [hunk 3]
```diff
@@ -1,37 +1,35 @@
...
-        mongo_user = os.getenv("MONGO_INITDB_ROOT_PASSWORD", "admin")
-        mongo_password = os.getenv("MONGO_INITDB_ROOT_PASSWORD", "admin")
-        mongo_host = os.getenv("MONGODB_HOST", "localhost:27017")
-        url = f"mongodb://{mongo_user}:{mongo_password}@{mongo_host}"
-        self.client = MongoClient(url)
 
-
-class DBService(MongoClientCreator):
+class DBService:
     """Create database service used to communicate with database.
```
Comment (blankdots) [inline metadata_backend/database/db_services.py:26]:
```suggestion
    """Create service used for database communication.
```

File: metadata_backend/database/db_services.py (hunk 1 above)
Comment (otahontas) [reply metadata_backend/database/db_services.py:17]: yeah, sounds reasonable. I'll add it

File: metadata_backend/database/db_services.py (hunk 2 above)
Comment (otahontas) [reply metadata_backend/database/db_services.py:114]: yeah, maybe we could catch ConnectionFailure, OperationFailure and then other pymongo errors.

Retry mechanism sounds like a task for later, I'll add issue for that.

**Pull request merged**